zilencer/views.py

```diff
@@ -435,7 +438,7 @@ def remote_server_post_analytics(
                     ("realm", check_int),
                     ("event_time", check_float),
                     ("backfilled", check_bool),
-                    ("extra_data", check_none_or(check_string)),
+                    ("extra_data", check_union((check_none_or(check_string), check_dict()))),
```

Comment:
I kinda want to express this as a union of three things rather than the `check_none_or(check_string)` itself inside a union, but I don't know it's worth adding a `check_none` validator.

Reply:
Maybe `check_none_or(check_union(...))`?